### PR TITLE
Updated Documentation (API Changes, On-Prem. FPGAs, Distributed Metasim)

### PR DESCRIPTION
**Pull request by abejgonzalez**
See title

#### Related PRs / Issues

All PRs since 1.13.0

### Contributor Checklist
- [x] Is this PR's title suitable for inclusion in the changelog and have you added a `changelog:<topic>` label?
- [x] Did you add Scaladoc/docstring/doxygen to every public function/method?
- [x] Did you add at least one test demonstrating the PR?
- [x] Did you delete any extraneous prints/debugging code?
- [x] Did you state the UI / API impact?
- [x] Did you specify the Verilog / AGFI compatibility impact?

- [x] If applicable, did you regenerate and publicly share default AGFIs?

- [x] If applicable, did you apply the `ci:fpga-deploy` label?

- [x] If applicable, did you apply the `Please Backport` label?

### Reviewer Checklist (only modified by reviewer)
- [ ] Is the title suitable for inclusion in the changelog and does the PR have a `changelog:<topic>` label?
- [ ] Did you mark the proper release milestone?
- [ ] Did you check whether all relevant Contributor checkboxes have been checked?

### CODE REVIEW

File: .github/workflows/firesim-run-tests.yml
Comment (davidbiancolin) [inline .github/workflows/firesim-run-tests.yml:193]: Why'd you remove this?

File: docs/Advanced-Usage/Debugging-and-Profiling-on-FPGA/Printf-Synthesis.rst
Comment (davidbiancolin) [inline docs/Advanced-Usage/Debugging-and-Profiling-on-FPGA/Printf-Synthesis.rst:83]: Might as well change this to use your DOCREF scheme, no?

File: docs/Advanced-Usage/Debugging-and-Profiling-on-FPGA/TracerV.rst
Comment (davidbiancolin) [inline docs/Advanced-Usage/Debugging-and-Profiling-on-FPGA/TracerV.rst:67]: I guess there are many examples of this :(. don't let the DOCREF stuff block the PR then

File: .github/workflows/firesim-run-tests.yml
Comment (davidbiancolin) [reply .github/workflows/firesim-run-tests.yml:193]: Ah -- so if paths move the doc build still happens. Nice.

File: docs/Advanced-Usage/Manager/Manager-Configuration-Files.rst
Comment (davidbiancolin) [inline docs/Advanced-Usage/Manager/Manager-Configuration-Files.rst:315]: You might be able to provide the gh ref here to a sample?  Also "build farm name" -> "build farm definition"?

File: docs/Advanced-Usage/Manager/Manager-Configuration-Files.rst
Comment (davidbiancolin) [inline docs/Advanced-Usage/Manager/Manager-Configuration-Files.rst:316]:
```suggestion
For example, if you want to run your FPGA builds on a pre-specified build farm that automatically spawns an EC2 instance per FPGA build, you can
```

File: docs/Advanced-Usage/Manager/Manager-Configuration-Files.rst
Comment (davidbiancolin) [inline docs/Advanced-Usage/Manager/Manager-Configuration-Files.rst:390]:
```suggestion
This field specifies the name of the python `build farm class` that implements your desired build farm.
```
Remove the parenthetical on the next line, it confused me.

File: docs/Advanced-Usage/Manager/Manager-Configuration-Files.rst
Comment (davidbiancolin) [inline docs/Advanced-Usage/Manager/Manager-Configuration-Files.rst:406]:
```suggestion
There are two mainline build farm configurations provided (their build farm classes are defined in :gh-file-ref:`deploy/buildtools/buildfarm.py`).
```

File: docs/Advanced-Usage/Manager/Manager-Configuration-Files.rst
Comment (davidbiancolin) [inline docs/Advanced-Usage/Manager/Manager-Configuration-Files.rst:422]:
```suggestion
Generally, running on a ``z1d.2xlarge`` is sufficient. For very large designs, using a z1d.3xlarge may be required to handle Vivado's memory footprint.`
```
You'll have to clean up the next line.

Comment (davidbiancolin) [reply docs/Advanced-Usage/Manager/Manager-Configuration-Files.rst:422]: It's still worth saying higher frequency instances may not help that much.

File: docs/Advanced-Usage/Manager/Manager-Configuration-Files.rst
Comment (davidbiancolin) [inline docs/Advanced-Usage/Manager/Manager-Configuration-Files.rst:426]:
```suggestion
``spot`` or ``ondemand``.
```

File: docs/Advanced-Usage/Manager/Manager-Configuration-Files.rst
Comment (davidbiancolin) [inline docs/Advanced-Usage/Manager/Manager-Configuration-Files.rst:435]: Can you explain why "default" is included in the name? It suggests there may be reasons it is put elsewhere.

File: docs/Advanced-Usage/Manager/Manager-Configuration-Files.rst
Comment (davidbiancolin) [inline docs/Advanced-Usage/Manager/Manager-Configuration-Files.rst:449]:
```suggestion
or on a set of machines they maintain.
```

File: docs/Advanced-Usage/Manager/Manager-Configuration-Files.rst
Comment (davidbiancolin) [inline docs/Advanced-Usage/Manager/Manager-Configuration-Files.rst:458]:
```suggestion
The ``build_farm_hosts`` argument section provides a list of IP addresses that
```
The "and" was confusing and your example covers it.

Comment (davidbiancolin) [reply docs/Advanced-Usage/Manager/Manager-Configuration-Files.rst:458]: also you probably want to use "host names" in place of "IP addresses" in this section.

File: docs/Advanced-Usage/Manager/Manager-Configuration-Files.rst
Comment (davidbiancolin) [inline docs/Advanced-Usage/Manager/Manager-Configuration-Files.rst:556]:
```suggestion
Naming rules: this must be all lowercase and you should stick to letters and numbers ([a-z0-9]).
```
Good to be explicit here.

File: docs/index.rst
Comment (davidbiancolin) [inline docs/index.rst:4]: We might want to remind ourselves to delete this. I guess it's on the front page.

File: docs/Advanced-Usage/Debugging-and-Profiling-on-FPGA/TracerV.rst
Comment (abejgonzalez) [reply docs/Advanced-Usage/Debugging-and-Profiling-on-FPGA/TracerV.rst:67]: Yea. I'll leave this to future work.

File: docs/Advanced-Usage/Manager/Manager-Configuration-Files.rst
Comment (abejgonzalez) [reply docs/Advanced-Usage/Manager/Manager-Configuration-Files.rst:315]: I'll change this to instead refer to the doc section on build farms. Additionally, I think "build farm name" -> "build farm configuration" makes more sense here.

File: docs/Advanced-Usage/Manager/Manager-Configuration-Files.rst
Comment (abejgonzalez) [reply docs/Advanced-Usage/Manager/Manager-Configuration-Files.rst:422]: Help in speeding up builds correct?

File: docs/Advanced-Usage/Manager/Manager-Configuration-Files.rst
Comment (abejgonzalez) [reply docs/Advanced-Usage/Manager/Manager-Configuration-Files.rst:435]: Technically, the "default_" part is not needed. The build directory on the build host will always be located by this argument. I added "default_" here because Sagar wanted the names to be consistent between the EC2/Localhost build farms.

Comment (abejgonzalez) [reply docs/Advanced-Usage/Manager/Manager-Configuration-Files.rst:435]: I added an extra sentence mentioning that this can't be overridden.

**Review by sagark — APPROVED** [review]
sure

**Pull request merged**
